azure/datalake/store/core.py

```diff
@@ -107,73 +107,101 @@ def open(self, path, mode='rb', blocksize=2**25, delimiter=None):
         return AzureDLFile(self, AzureDLPath(path), mode, blocksize=blocksize,
                            delimiter=delimiter)
 
-    def _ls(self, path):
+    def _ls(self, path, invalidate_cache=True):
         """ List files at given path """
         path = AzureDLPath(path).trim()
         key = path.as_posix()
+        
+        if invalidate_cache:
+            self.invalidate_cache(key)
+
         if key not in self.dirs:
             out = self.azure.call('LISTSTATUS', key)
             self.dirs[key] = out['FileStatuses']['FileStatus']
             for f in self.dirs[key]:
                 f['name'] = (path / f['pathSuffix']).as_posix()
         return self.dirs[key]
 
-    def ls(self, path="", detail=False):
+    def ls(self, path="", detail=False, invalidate_cache=True):
```

Comment:
Can't wait until you get to drop Python 2.7 support as this would have made a great keyword-only argument.